### PR TITLE
Limit interfaces listened on by NGINX, fixes #54

### LINKED ISSUE #54: It's still possible to access services exposed by other clients
Server uses well-known port ranges to create nginx listeners for proxied services. Those listeners can be accessed by other (rogue) clients from the wireguard network.

We should ensure, that respective listeners are created only on interfaces they were destined to work with.

Example from the devenv. Expose nginx service, open shell in nginx container in any of the clients and shoot:
```
/ $ curl 10.188.0.1:25001
/ $ curl 10.188.0.1:25002
```

It accesses nginx services using tunnells to various clients.

### PR DESCRIPTION
**Pull request by glothriel**

### CODE REVIEW

**Comment by glothriel** [comment]
Missing test for checking the connection through the tunnel - apparently something's broken and nothing indicates that

File: tests/test_kubernetes.py [hunk 1]
```diff
@@ -201,4 +201,44 @@ def _ensure_that_proxied_service_is_deleted():
...
+    )
+    annotator.do("wormhole.glothriel.github.com/exposed", "yes")
+
+    @retry(tries=60, delay=1)
+    def _ensure_that_proxied_service_is_created():
+        assert (
+            len(kubectl.json(["-n", "server", "get", "svc"])["items"])
+            == amount_of_services_before_annotation + 1
+        )
+    _ensure_that_proxied_service_is_created()
```
Comment (jzyinq) [inline tests/test_kubernetes.py:228]: Looks like a candidate another fixture/helper - you have the same thingie in `test_changing_annotation_causes_creating_proxy_service`

`expose_service(annotator)`

File: pkg/nginx/listener_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+type mockLister struct {
+	interfaces []networkInterface
+	err        error
+}
+
+func (m *mockLister) Interfaces() ([]networkInterface, error) {
+	return m.interfaces, m.err
+}
+
+func TestAllAcceptWgListener(t *testing.T) {
```
Comment (jzyinq) [inline pkg/nginx/listener_test.go:18]: What about other listener types? `portOnlyListener` / `givenAddressOnlyListener`

File: pkg/nginx/listener_test.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		interfaces: []networkInterface{
+			{
+				name:      "eth0",
+				addresses: []string{"127.0.0.1"},
+			},
+			{
+				name:      "wg0",
+				addresses: []string{"10.178.2.1"},
+			},
+		},
```
Comment (jzyinq) [inline pkg/nginx/listener_test.go:30]: I'm missing test cases for those edges: 
```go 
interfaces, interfacesErr := a.listener.Interfaces()
if interfacesErr != nil {
	return []string{}, interfacesErr
}
```

```go
if len(allAddrs) == 0 {
	return []string{}, errors.New("No network interfaces matching conditions found")
}
```

File: pkg/nginx/listener.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+func (a *allAcceptWg0Listener) Addrs(portNumber int) ([]string, error) {
+	interfaces, interfacesErr := a.lister.Interfaces()
+	if interfacesErr != nil {
+		return []string{}, interfacesErr
+	}
+
+	var allAddrs []string
+
+	for _, iface := range interfaces {
+		if iface.name == "wg0" {
```
Comment (jzyinq) [inline pkg/nginx/listener.go:52]: `const wg0If` perhaps?

File: pkg/nginx/listener.go [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+		address: address,
+	}
+}
+
+type defaultNetworkInterfaceLister struct {
+}
+
+// Interfaces implements networkinterfacelister
+func (l *defaultNetworkInterfaceLister) Interfaces() ([]networkInterface, error) {
+	ifaces, err := net.Interfaces()
```
Comment (jzyinq) [inline pkg/nginx/listener.go:101]: Maybe extract ifaces as struct attribute? That way you could test out this part of code as well.

File: tests/test_kubernetes.py (hunk 1 above)
Comment (glothriel) [reply tests/test_kubernetes.py:228]: I added a `Service` util with a classmethod for getting amount of services in given ns and list of service names. I don't want the tests to hide too much, they should be verbose enough so you immediately know what the test is for.

File: pkg/nginx/listener_test.go (hunk 2 above)
Comment (glothriel) [reply pkg/nginx/listener_test.go:18]: removed portOnly and wrote test for givenAddressOnlyListener

File: pkg/nginx/listener_test.go (hunk 3 above)
Comment (glothriel) [reply pkg/nginx/listener_test.go:30]: Added

File: pkg/nginx/listener.go (hunk 4 above)
Comment (glothriel) [reply pkg/nginx/listener.go:52]: Added

File: pkg/nginx/listener.go (hunk 5 above)
Comment (glothriel) [reply pkg/nginx/listener.go:101]: This whole struct was extracted out of `Listener` interface to test it. Looking on amount of mocks I'd have to write to test this i think it's enough abstraction

**Review by jzyinq — APPROVED** [review]

**Pull request merged**